ws/src/can_controller/src/velocity_to_can.cpp

```diff
@@ -0,0 +1,97 @@
+#include <string>
+#include <tuple>
+
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/string.hpp"
+#include "geometry_msgs/Twist.h"
+
+// TODO: Install the following libraries:
+#include "ctre/Phoenix.h"
+#include "ctre/phoenix/platform/Platform.h"
+#include "ctre/phoenix/unmanaged/Unmanaged.h"
+#include "ctre/phoenix/cci/Unmanaged_CCI.h"
+
+using namespace std::chrono_literals;
+using namespace ctre::phoenix;
+using namespace ctre::phoenix::platform;
+using namespace ctre::phoenix::motorcontrol;
+using namespace ctre::phoenix::motorcontrol::can;
+
+// TODO: Update these port numbers!
+TalonSRX talLeft(1);
+TalonSRX talRght(0);
+
+talLeft.ConfigSelectedFeedbackSensor(...); // TODO: finish filling out, though it may be unnecessary.
+
+int kPIDLoopIdx = 0;
+int kTimeoutMs = 30;
+
+talLeft.Config_kF(kPIDLoopIdx, 0, kTimeoutMs); // TODO: Figure out PIDF values.
+talLeft.Config_kP(kPIDLoopIdx, 0, kTimeoutMs); // TODO: this might have to be moved to main, alongside the ConfigSelectedFeedbackSensor()
+talLeft.Config_kI(kPIDLoopIdx, 0, kTimeoutMs);
+talLeft.Config_kD(kPIDLoopIdx, 0, kTimeoutMs);
+
+// TODO: we will need to tune the velocity PIDF values according to https://phoenix-documentation.readthedocs.io/en/latest/ch16_ClosedLoop.html, probably.
+
+void initDrive()
+{
+	talRght.SetInverted(false); /* both talons should blink green when driving forward */
+}
+
+void drive(double left, double right)
+{
+	talLeft.Set(ControlMode::Velocity, left); // TODO: Update controlmode to be accurate
+	talRght.Set(ControlMode::Velocity, right);
+}
+
+class TalonNode : public rclcpp::Node
+{
+  public:
+    TalonNode()
+    : Node("talon_interface") // TODO: Replace node name
+    {
+      subscription_ = this->create_subscription<tuple>("PLACEHOLDER_VELOCITY_TOPIC", 10, std::bind(&PathfinderSubscriber::topic_callback, this, _1));
+      left_encoder_publisher_ = this->create_publisher<geometry_msgs::Twist>("cmd_vel/left_wheel_encoder_position", 10)
+      right_encoder_publisher_ = this->create_publisher<geometry_msgs::Twist>("cmd_vel/right_wheel_encoder_position", 10)
+      timer_ = this->create_wall_timer(20ms, std::bind(&TalonNode::timer_callback, this)); //TODO: Change timer time.
+    }
+  //TODO: WARNING—CURRENTLY, THIS AUTOMATICALLY RUNS THE MOTORS WHENEVER THIS NODE IS RUNNING... JUST FYI.
+  private:
+    void topic_callback(const tuple::SharedPtr msg) const
+    {
+      double left_vel = get<0>msg;
+      double right_vel = get<1>msg;
+      drive(left_vel, right_vel);
+    }
+    
+    void timer_callback()
+    {
+      geometry_msgs::Twist left_encoder_out;
+      geometry_msgs::Twist right_encoder_out;
+      left_encoder_out.linear.x = talLeft.getSelectedSensorVelocity();
+      right_encoder_out.linear.x = talRight.getSelectedSensorVelocity();
+      left_encoder_publisher_->publish(left_encoder_out);
+      right_encoder_publisher_->publish(right_encoder_out);
+    }
+    rclcpp::Subscription<tuple>::SharedPtr subscription_;
```

Comment:
Also what's up with the `_` at the end of these private variables? Is that the standard set by `cpplint`?

Reply:
Standard in ROS demos, though I'm happy to flexible if we have a preferred style guide.

Reply:
@Michael-Equi What's the preferred style for private variables? There might be a linter in this project as well